src/DGmethods/DGmodel.jl

```diff
@@ -8,124 +8,110 @@ end
-  Nfp = Nq * Nqk
-  nrealelem = length(topology.realelems)
-
-  Qvisc = param.diff
-  auxstate = param.aux
+  Ω    = grid.topology.realelems
+  nΩ   = length(Ω)
+  Nᵈ   = dimensionality(grid)
+  N    = polynomialorder(grid)
+  Nq   = N + 1
+  Nqk  = Nᵈ == 2 ? 1 : Nq
+  Nfp  = Nq * Nqk
+  Np   = dofs_per_element(grid)
+
+  σ  = param.diff
+  α  = param.aux
+  nσ = num_diffusive(bl, DFloat)
 
-  DFloat = eltype(Q)
-  nviscstate = num_diffusive(bl, DFloat)
-
-  lgl_weights_vec = grid.ω
-  Dmat = grid.D
   vgeo = grid.vgeo
   sgeo = grid.sgeo
-  vmapM = grid.vmapM
-  vmapP = grid.vmapP
-  elemtobndy = grid.elemtobndy
-  polyorder = polynomialorder(dg.grid)
-
-  Np = dofs_per_element(grid)
-
-  if hasmethod(update_aux!, Tuple{typeof(dg), typeof(bl), typeof(Q),
-                                  typeof(auxstate), typeof(t)})
-    update_aux!(dg, bl, Q, auxstate, t)
+  ω    = grid.ω
+  D    = grid.D
+  ι⁻   = grid.vmapM
+  ι⁺   = grid.vmapP
+  ιᴮ   = grid.elemtobndy
+
+  if hasmethod(update_aux!, Tuple{typeof(dg), typeof(bl), typeof(Q), typeof(α),
+                                  typeof(t)})
+    update_aux!(dg, bl, Q, α, t)
   end
 
   ########################
   # Gradient Computation #
   ########################
   MPIStateArrays.start_ghost_exchange!(Q)
-  MPIStateArrays.start_ghost_exchange!(auxstate)
+  MPIStateArrays.start_ghost_exchange!(α)
 
-  if nviscstate > 0
-
-    @launch(device, threads=(Nq, Nq, Nqk), blocks=nrealelem,
-            volumeviscterms!(bl, Val(dim), Val(polyorder), Q.Q, Qvisc.Q, auxstate.Q, vgeo, t, Dmat,
-                             topology.realelems))
+  if nσ > 0
+    @launch(device, threads=(Nq, Nq, Nqk), blocks=nΩ,
+            volumeviscterms!(bl, Val(Nᵈ), Val(N), Q.Q, σ.Q, α.Q, vgeo, t, D, Ω))
 
     MPIStateArrays.finish_ghost_recv!(Q)
-    MPIStateArrays.finish_ghost_recv!(auxstate)
+    MPIStateArrays.finish_ghost_recv!(α)
 
-    @launch(device, threads=Nfp, blocks=nrealelem,
-            faceviscterms!(bl, Val(dim), Val(polyorder), dg.gradnumflux,
-                           Q.Q, Qvisc.Q, auxstate.Q,
-                           vgeo, sgeo, t, vmapM, vmapP, elemtobndy,
-                           topology.realelems))
+    @launch(device, threads=Nfp, blocks=nΩ,
+            faceviscterms!(bl, Val(Nᵈ), Val(N), dg.gradnumflux, Q.Q, σ.Q, α.Q,
+                           vgeo, sgeo, t, ι⁻, ι⁺, ιᴮ, Ω))
 
-    MPIStateArrays.start_ghost_exchange!(Qvisc)
+    MPIStateArrays.start_ghost_exchange!(σ)
   end
 
   ###################
   # RHS Computation #
   ###################
-  @launch(device, threads=(Nq, Nq, Nqk), blocks=nrealelem,
-          volumerhs!(bl, Val(dim), Val(polyorder), dQdt.Q, Q.Q, Qvisc.Q, auxstate.Q,
-                     vgeo, t, lgl_weights_vec, Dmat,
-                     topology.realelems, increment))
+  @launch(device, threads=(Nq, Nq, Nqk), blocks=nΩ,
+          volumerhs!(bl, Val(Nᵈ), Val(N), dQdt.Q, Q.Q, σ.Q, α.Q, vgeo, t, ω, D,
+                     Ω, increment))
```

Comment:
I think it's better if we replace rhs by tendency to avoid confusion (a discussion we had a few times in the past...)

Reply:
makes sense to me

do we want something like volume_tendency! or volume_tendency_contribution! etc?

Reply:
okay changes are 

`volumerhs!` --> `volume_tendency!`
`facerhs!` --> `face_tendency!`